netket/graph/space_group.py

```diff
@@ -79,25 +79,145 @@ def _ensure_iterable(x):
         return x
 
 
+def _translations_along_axis(lattice: Lattice, axis: int) -> PermutationGroup:
+    """
+    The group of valid translations along an axis as a `PermutationGroup`
+    acting on the sites of `lattice.`
+    """
+    if lattice._pbc[axis]:
+        trans_list = [Identity()]
+        # note that we need the preimages in the permutation
+        trans_perm = lattice.id_from_position(
+            lattice.positions - lattice.basis_vectors[axis]
+        )
+        vector = np.zeros(lattice.ndim, dtype=int)
+        vector[axis] = 1
+        trans_by_one = Translation(trans_perm, vector)
+
+        for _ in range(1, lattice.extent[axis]):
+            trans_list.append(trans_list[-1] @ trans_by_one)
+
+        return PermutationGroup(trans_list, degree=lattice.n_nodes)
+    else:
+        return PermutationGroup([Identity()], degree=lattice.n_nodes)
+
+
+def _pg_to_permutation(lattice: Lattice, point_group: PointGroup) -> PermutationGroup:
+    """
+    The permutation action of `point_group` on the sites of `lattice`.
+    """
+    perms: list[Element] = []
+    for p in point_group:
+        if isinstance(p, Identity):
+            perms.append(Identity())
+        else:
+            # note that we need the preimages in the permutation
+            perm = lattice.id_from_position(p.preimage(lattice.positions))
+            perms.append(Permutation(perm, name=str(p)))
+    return PermutationGroup(perms, degree=lattice.n_nodes)
+
+
+@struct.dataclass
+class TranslationGroup(PermutationGroup):
+    """
+    Class to handle translation symmetries of a `Lattice`.
+
+    Can be used as a `PermutationGroup` representing the translations,
+    but the product table is computed much more efficiently than a generic
+    `PermutationGroup`.
+    """
+
+    lattice: Lattice
+    """The lattice whose translation group is represented."""
+    axes: tuple[int]
+    """Axes translations along which are represented by the group."""
+    group_shape: tuple[int]
+    """Size of the translation group along each axis"""
```

Comment:
We do not need to keep this, right? 
It can be computed from `lattice` and `axes`.
To minimise the risk of bugs, where this would get 'out of sync' or be set erroneously, I would just make it a property. 

I would not keep this as an attribute, and instead have it as a property (you can also cache it if you want and if you think it's expensive to compute).

Reply:
something like
```python
@property # or @struct.property_cached
def group_shape(self):
        translation_by_axis = [_translations_along_axis(self.lattice, i) for i in self.axes]
        return tuple(len(x) for x in translation_by_axis)
```

Reply:
It actually follows from `lattice.pbc` and `lattice.extent` so it's quite cheap to compute!